hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -406,6 +407,9 @@ public void onResponse(Data response) {
                 public void onFailure(Throwable t) {
                 }
             });
+            if (getMapConfig().isStatisticsEnabled()) {
```

Comment:
The problem in your case is that you have to have the start time; you are now using the callback as a container for the start time so you can calculate the difference on completion. 

The problem is that map was not designed for speed and statistics are enabled by default, but quite expensive; also in memory usage since additional object + fields are tracked. Making the memory overhead by default larger than needs to be.